### PR TITLE
Good gravy, it's a new food (and drinks) update!

### PR DESCRIPTION
**Pull request by tomixcomics**
This update adds:
-6 new food dishes (elotes, steak tartare, biscuits and gravy, and 3 fancy grilled cheeses) (4 of which were special requests made by other players)
-Gravy and gravy boats
-New sprites for mashed potatoes and schnitzels that had gravy poured on top of them.
-2 new drinks (dyn boba tea, chocolate soda)
-New sprite for blizzard ear flour boxes because it was really confusing they had the same sprite as regular flour.

All created and sprited by me.

### CODE REVIEW

**Comment by tomixcomics** [comment]
!review

File: code/modules/cooking/plates.dm [hunk 1]
```diff
@@ -186,3 +186,43 @@ Plates that can hold your cooking stuff
...
+/obj/item/reagent_containers/bowl/gravy_boat/afterattack(var/obj/target, var/mob/user, var/flag)
+	if(!target.is_open_container() || !flag)
+		return ..(target, user, flag)
+	if(reagents.total_volume)
+		if(!target.reagents || !REAGENTS_FREE_SPACE(target.reagents))
+			to_chat(user, SPAN_NOTICE("[target] is full."))
+			return TRUE
+		var/trans = reagents.trans_to(target, amount_per_transfer_from_this) //sprinkling reagents on generic non-mobs
+		user.visible_message(
+			"<b>[user]</b> pours onto [target] from [src].",
```
Comment (FluffyGhoster) [inline code/modules/cooking/plates.dm:213]: SPAN_BOLD

File: code/modules/cooking/plates.dm [hunk 2]
```diff
@@ -186,3 +186,43 @@ Plates that can hold your cooking stuff
...
+			SPAN_NOTICE("You transfer [trans] units of the solution.")
+		)
+		playsound(user.loc, /singleton/sound_category/generic_pour_sound, 10, 1)
```
Comment (FluffyGhoster) [inline code/modules/cooking/plates.dm:216]: No magic numbers

**Review by alsoandanswer — APPROVED** [review]

File: code/modules/cooking/plates.dm (hunk 2 above)
Comment (tomixcomics) [reply code/modules/cooking/plates.dm:216]: idk what magic numbers are i copied this code from somewhere else ._.

File: code/modules/cooking/plates.dm (hunk 1 above)
Comment (tomixcomics) [reply code/modules/cooking/plates.dm:213]: no idea what this means o-o
this is code i reused form somewhere else

File: code/modules/cooking/plates.dm (hunk 2 above)
Comment (FluffyGhoster) [reply code/modules/cooking/plates.dm:216]: https://en.wikipedia.org/wiki/Magic_number_(programming)

```
playsound(user.loc, /singleton/sound_category/generic_pour_sound, 10, TRUE)
```

File: code/modules/cooking/plates.dm (hunk 1 above)
Comment (FluffyGhoster) [reply code/modules/cooking/plates.dm:213]: Like the line below, you have to use SPAN_BOLD instead of raw html

Comment (tomixcomics) [reply code/modules/cooking/plates.dm:213]: So would this be right?

		var/trans = reagents.trans_to(target, amount_per_transfer_from_this) //sprinkling reagents on generic non-mobs
		user.visible_message(
			"[SPAN_BOLD(user)] pours onto [target] from [src]"
			SPAN_NOTICE("You transfer [trans] units of the solution.")
		)
		playsound(user.loc, /singleton/sound_category/generic_pour_sound, 10, TRUE)

Comment (SleepyGemmy) [inline code/modules/cooking/plates.dm:213]:
```suggestion
			SPAN_BOLD("[user]") + "pours onto \the [target] from \the [src].",
```

File: code/modules/cooking/plates.dm (hunk 2 above)
Comment (SleepyGemmy) [inline code/modules/cooking/plates.dm:216]:
```suggestion
		playsound(get_turf(user), /singleton/sound_category/generic_pour_sound, 10, TRUE)
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 3]
```diff
@@ -419,6 +419,8 @@
 	taste_description = "chalky starch"
 	color = "#DFDEA1"
 	condiment_name = "Adhomian flour sack"
+	condiment_desc = "Flour ground from pure Adhomian Blizzard Ears!"
```
Comment (SleepyGemmy) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:422]:
```suggestion
	condiment_desc = "Flour ground from pure Adhomian blizzard ears!"
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 4]
```diff
@@ -6986,6 +6988,17 @@
...
 
+/singleton/reagent/drink/dyn_boba
+	name = "Dyn Boba Tea"
+	description = "Dyn ice tea with boba pearls."
+	nutrition = 3
+	color = "#00ecec"
+	taste_description = "fizzy mint tapioca tea"
+	glass_icon_state = "boba_dyn"
+	glass_name = "dyn boba"
+	glass_desc = "A dyn boba drink, sometimes reffered to as 'Boba Dyn', combining both Skrell and Human cultures into their natural result of putting tasty, soft, chewy things in drinks."
```
Comment (SleepyGemmy) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:6999]:
```suggestion
	glass_desc = "A dyn boba drink, sometimes referred to as 'boba dyn', combining both skrell and human cultures of putting tasty, soft, chewy things in drinks."
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 5]
```diff
@@ -6995,3 +7008,22 @@
 	glass_icon_state = "zobo_glass"
 	glass_name = "glass of zobo"
 	glass_desc = "A roselle juice popular across Eridani, often drunk cold."
+
+/singleton/reagent/condiment/gravy
+	name = "Gravy"
+	description = "A thick sauce made from the juice of meat that occurs naturally during cooking."
+	taste_description = "gravy"
+	color = "#a36a35"
+	condiment_desc = "A thick meaty suace for your dishes."
```
Comment (SleepyGemmy) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:7017]:
```suggestion
	condiment_desc = "A thick meaty sauce for your dishes."
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 6]
```diff
@@ -4528,3 +4528,25 @@
...
 	result_amount = 6
+
+/datum/chemical_reaction/drink/dyn_boba
+	name = "dyn boba"
+	id = "boba_dyn"
+	result = /singleton/reagent/drink/dyn_boba
+	required_reagents = list(/singleton/reagent/drink/dynjuice/cold = 5, /singleton/reagent/drink/boba = 1)
+	result_amount = 6
+
+//
```
Comment (SleepyGemmy) [inline code/modules/reagents/Chemistry-Recipes.dm:4539]:
```suggestion
```

File: code/modules/reagents/reagent_containers/food/snacks/breads.dm [hunk 7]
```diff
@@ -510,6 +510,39 @@
 	reagent_data = list(/singleton/reagent/nutriment = list("toasted bread" = 3, "cheese" = 3))
 	bitesize = 2
 
+/obj/item/reagent_containers/food/snacks/grilled_mac_and_cheese
+	name = "grilled mac and cheese sandwich"
+	desc = "You know what would make these carbs better? MORE CARBS. Gooey macaroni and cheese between two careful slices of crunchy, toasted bread. This was either invented by a hipster, a broke college student, or both."
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/breads.dm:515]:
```suggestion
	desc = "You know what would make these carbs better? MORE CARBS! Gooey macaroni and cheese between two careful slices of crunchy, toasted bread. This was either invented by a hipster, a broke college student, or both."
```

File: code/modules/reagents/reagent_containers/food/snacks/breads.dm [hunk 8]
```diff
@@ -510,6 +510,39 @@
...
+	icon_state = "maccheesesandwich"
+	trash = /obj/item/trash/plate
+	filling_color = "#D9BE29"
+	reagents_to_add = list(/singleton/reagent/nutriment/protein = 1)
+	reagent_data = list(/singleton/reagent/nutriment/protein = list("cheese" = 5))
+	bitesize = 3
+
+/obj/item/reagent_containers/food/snacks/grilled_triple_cheese_crunch_sandwich
+	name = "grilled triple cheese crunch sandwich"
+	desc = "Who needs a four cheese medley when you have the classy and playful variety of Cheddar, Nakarka, and uh... Whatever that glowy orange crud they put in cheesie honkers is, all in one sandwich?"
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/breads.dm:526]:
```suggestion
	desc = "Who needs a four cheese medley when you have the classy and playful variety of cheddar, nakarka, and uh... whatever that glowy orange crud they put in cheesie honkers is, all in one sandwich?"
```

File: code/modules/reagents/reagent_containers/food/snacks/breads.dm [hunk 9]
```diff
@@ -510,6 +510,39 @@
...
+	icon = 'icons/obj/item/reagent_containers/food/bread.dmi'
+	icon_state = "triplecheesecrunch"
+	trash = /obj/item/trash/plate
+	filling_color = "#b9d929"
+	reagents_to_add = list(/singleton/reagent/nutriment = 5, /singleton/reagent/nutriment/protein = 5)
+	reagent_data = list(/singleton/reagent/nutriment = list("toasted bread" = 2, "tangy cheese" = 3), /singleton/reagent/nutriment/protein = list("sharp cheese" = 3,"crunchy cheese" = 3))
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/breads.dm:532]:
```suggestion
	reagent_data = list(/singleton/reagent/nutriment = list("toasted bread" = 2, "tangy cheese" = 3), /singleton/reagent/nutriment/protein = list("sharp cheese" = 3, "crunchy cheese" = 3))
```

File: code/modules/reagents/reagent_containers/food/snacks/cultural/human.dm [hunk 10]
```diff
@@ -244,6 +244,43 @@
...
+	switch(percent_biscuits_and_gravy)
+		if(0 to 49)
+			icon_state = "gravybiscuits_half"
+		if(50 to INFINITY)
+			icon_state = "gravybiscuits"
+
+/obj/item/reagent_containers/food/snacks/elotes
+	name = "elotes"
+	gender = PLURAL
+	desc = "Grilled mexican sweet corn with chili powder, mayonnaise, cheese, sour cream and seasonings."
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/cultural/human.dm:274]:
```suggestion
	desc = "Grilled mexican sweet corn with chili powder, mayonnaise, cheese, sour cream, and seasonings."
```

File: code/modules/reagents/reagent_containers/food/snacks/cultural/human.dm [hunk 11]
```diff
@@ -244,6 +244,43 @@
...
+	icon = 'icons/obj/item/reagent_containers/food/cultural/human.dmi'
+	icon_state = "elotes"
+	trash = /obj/item/trash/plate
+
+	reagents_to_add = list(/singleton/reagent/nutriment = 6, /singleton/reagent/nutriment/protein = 3)
+	reagent_data = list(/singleton/reagent/nutriment = list("corn" = 5, "zesty seasoning" = 3, "lime" = 3), /singleton/reagent/nutriment/protein = list("cheese" = 5)) //please feel free to correct this if it's inaccurate, i have never had an elote.
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/cultural/human.dm:280]:
```suggestion
	reagent_data = list(/singleton/reagent/nutriment = list("corn" = 5, "zesty seasoning" = 3, "lime" = 3), /singleton/reagent/nutriment/protein = list("cheese" = 5))
```

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 12]
```diff
@@ -448,3 +454,23 @@
 			icon_state = "cozmo_cubes_half"
 		if(70 to INFINITY)
 			icon_state = "cozmo_cubes"
+
+/obj/item/reagent_containers/food/snacks/steak_tartare
+	name = "steak tartare"
+	desc = "Excuse me! Waiter! Is this some kind of joke?! This is raw beef you just SLAPPED onto a plate and sprinkled with condiments! It is COMPLETELY RAW! Even the EGG isn't cooked! Are you kidding me?! Are you trying to give me food poisoning?! What kind of scam are you running here?! No, don't give me that 'it's supposed to be like this' nonsense, the menu said STEAK! Take this back and cook the hell out of it until it stops mooing at me!"
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm:460]:
```suggestion
	desc = "Raw beef sprinkled with condiments. Completely raw, including the egg. Hope it doesn't cause food poisoning."
```
while creative descriptions are fine, this is a bit over the top. i wrote a shorter variant that should keep the spirit of the longer variant, but feel free to write your own.

File: html/changelogs/Tomixcomics-PR-cheesyfoods.yml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, this gets changed to [] after reading.  Just remove the brackets when you add new shit.
+# Please surround your changes in  double quotes ("). It works without them, but if you use certain characters it screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Added 5 new food dishes: 3 grilled cheese sandwich varieties, elotes, biscuits and gravy, and steak tartare."
+  - rscadd: "Added 2 new drinks: Dyn boba tea and chocolate soda."
+  - rscadd: "Added gravy and gravy boats, along with sprite variants for mashed potatoes and schnitzels that had gravy poured on them."
+  - imageadd: "Added a new sprite for Blizzard ears flour boxes."
```
Comment (SleepyGemmy) [inline html/changelogs/Tomixcomics-PR-cheesyfoods.yml:61]:
```suggestion
  - rscadd: "Added 2 new drinks: dyn boba tea and chocolate soda."
  - rscadd: "Added gravy and gravy boats, along with sprite variants for mashed potatoes and schnitzels that had gravy poured on them."
  - imageadd: "Added a new sprite for blizzard ears flour boxes."
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
coffee jar sprite was changed a bit, please mention it in the changelog.

File: code/modules/cooking/plates.dm [hunk 14]
```diff
@@ -186,3 +186,43 @@ Plates that can hold your cooking stuff
...
+	name = "gravy boat"
+	desc = "Let's sail the seas of deliciousness!"
+	icon_state = "gravy"
+	amount_per_transfer_from_this = 5
+	possible_transfer_amounts = list(5, 10, 30)
+	w_class = WEIGHT_CLASS_SMALL
+	volume = 40
+	force = 14
+	atom_flags = ATOM_FLAG_OPEN_CONTAINER
+	item_flags = ITEM_FLAG_NO_BLUDGEON
```
[the remark this answers is not in the document]
Comment (tomixcomics) [reply code/modules/cooking/plates.dm:200]: I thought this was weird, but that's how it was phrased in the code I copied it from for some strange reason (I copied bowl code)

Also thank you kindly for all your edit suggestions!!!

**Comment by tomixcomics** [comment]
> coffee jar sprite was changed a bit, please mention it in the changelog.

I didn't touch anything that has to do with coffee jars, I'm not even sure which sprite you mean? Did I change something by accident? I can't find an item in the code called coffee jar so I'm not sure which sprite you're referring to.

Either way thank you kindly for your edits and taking the time to make them

**Comment by SleepyGemmy** [comment]
> I'm not even sure which sprite you mean?
image
image

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm (hunk 12 above)
Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/meat.dm]: Changing it to something else that's also short. thank you!

**Review by SleepyGemmy — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**